forest/benchmarking/tests/test_direct_fidelity_estimation.py

```diff
@@ -33,29 +42,52 @@ def test_exhaustive_dfe_run(benchmarker: BenchmarkConnection):
         assert expectation == 1.
 
 
-def test_monte_carlo_dfe(benchmarker: BenchmarkConnection):
+def test_exhaustive_state_dfe_run(benchmarker: BenchmarkConnection):
+    wfnsim = NumpyWavefunctionSimulator(n_qubits=1)
+    process = Program(X(0))
+    texpt = exhaustive_state_dfe(program=process, qubits=[0], benchmarker=benchmarker)
+    for setting in texpt.exp:
+        setting = setting[0]
+        prog = Program()
+        for oneq_state in setting.in_state.states:
+            prog += _one_q_state_prep(oneq_state)
+        prog += process
+
+        expectation = wfnsim.reset().do_program(prog).expectation(setting.out_operator)
+        assert expectation == 1.
+
+
+def test_monte_carlo_process_dfe(benchmarker: BenchmarkConnection):
     process = Program(CNOT(0, 1))
     texpt = monte_carlo_process_dfe(program=process, qubits=[0, 1], n_terms=10,
                                     benchmarker=benchmarker)
-    assert len(texpt) == 10
+    assert len(texpt.exp) == 10
 
     wfnsim = NumpyWavefunctionSimulator(n_qubits=2)
-    for setting in texpt:
+    for setting in texpt.exp:
         setting = setting[0]
         prog = Program()
         for oneq_state in setting.in_state.states:
             prog += _one_q_state_prep(oneq_state)
         prog += process
 
         expectation = wfnsim.reset().do_program(prog).expectation(setting.out_operator)
-        assert expectation == 1.
+        assert_almost_equal(expectation,1.,decimal=7)
+
 
+def test_monte_carlo_state_dfe(benchmarker: BenchmarkConnection):
+    process = Program(H(0), CNOT(0, 1))
+    texpt = monte_carlo_state_dfe(program=process, qubits=[0, 1], n_terms=10,
+                                    benchmarker=benchmarker)
+    assert len(texpt.exp) == 10
+
+    wfnsim = NumpyWavefunctionSimulator(n_qubits=2)
+    for setting in texpt.exp:
+        setting = setting[0]
+        prog = Program()
+        for oneq_state in setting.in_state.states:
+            prog += _one_q_state_prep(oneq_state)
+        prog += process
 
-def test_ratio_variance():
```

Comment:
@msohaibalam were these tests included in the pyQuil version?

Reply:
There are a few tests for `ratio_variance` specifically in the pyQuil version. See e.g. https://github.com/rigetti/pyquil/blob/master/pyquil/tests/test_operator_estimation.py#L556